fuel-core-interfaces/src/txpool.rs

```diff
@@ -327,16 +327,46 @@ pub enum TxPoolMpsc {
 pub enum TxStatus {
     /// Submitted into txpool.
     Submitted,
-    /// Executed in fuel block.
-    Executed,
+    /// Transaction has either been:
+    /// - successfully executed and included in a block.
+    /// - failed to execute and discarded.
```

Comment:
reverted transactions are still part of state.

```suggestion
    /// - failed to execute and state changes reverted
```